lib.sh

```diff
@@ -511,7 +511,21 @@ then
     mkdir -p "$3"
 fi
     rm -f "$3"/"$2"
-    curl -sfL "$1"/"$2" -o "$3"/"$2"
+    local retries=0
+    until [ "$retries" -ge 10 ]
+    do
+        if ! curl -sfL "$1"/"$2" -o "$3"/"$2"
+        then
+            msg_box "We just tried to fetch '$1/$2', but it seems like the server for the download isn't reachable, or that a temporary error occurred. We will now try again.
+
+Please report this issue to $ISSUES"
+            retries=$((retries+1))
+            print_text_in_color "$ICyan" "$retries of 10 retries."
+            countdown "Trying again in 30 seconds..." "30"
+        else
+            break
```

Comment:
@enoch85 we need to exit here, or at least return 1, no?

Reply:
No, this breaks if it works.

Reply:
You are right. But it continues after trying 10 times...

Reply:
Yes it does. As you said it might not be a good idea to just exit, which makes sense (but at the same time not, because if the file isn't there then the script will break anyway).

How should we handle this?

Reply:
See #2009 how I would handle that